components/mdns/mdns.c

```diff
@@ -3511,14 +3534,25 @@ void mdns_parse_packet(mdns_rx_packet_t *packet)
     const uint8_t *content = data + MDNS_HEAD_LEN;
     bool do_not_reply = false;
     mdns_search_once_t *search_result = NULL;
+    mdns_browse_t *browse_result = NULL;
+    char browse_result_hostname[MDNS_NAME_BUF_LEN] = {0};
+    char browse_result_instance[MDNS_NAME_BUF_LEN] = {0};
+    char browse_result_service[MDNS_NAME_BUF_LEN] = {0};
+    char browse_result_proto[MDNS_NAME_BUF_LEN] = {0};
```

Comment:
This may take a lot of stack space. Could we allocate dynamically only if we're browsing (or maybe ifdef it out if browsing not enabled)?

Reply:
Changed the logic, now using `malloc` to allocate the memory when used.